### PR TITLE
Add index pages to support heading clickability

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Add index pages to support heading clickability

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _query-dsl/index.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+layout: default
+title: Query DSL, aggregations, and analyzers
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+# Query DSL, aggregations, and analyzers
+
+[Analyzers]({{site.url}}{{site.baseurl}}/analyzers/text-analyzers/) process text to make it searchable. OpenSearch provides various analyzers, which let you customize the way text is split into terms and converted into a structured format. To search documents written in a different language, you can use one of the built-in [language analyzers]({{site.url}}{{site.baseurl}}/query-dsl/analyzers/language-analyzers/) for your language of choice.
```
Comment (vagimeli) [inline _query-dsl/index.md:11]:
```suggestion
[Analyzers]({{site.url}}{{site.baseurl}}/analyzers/text-analyzers/) process text to make it searchable. OpenSearch provides various analyzers, which let you customize the way text is split into terms and is converted into a structured format. To search documents written in a different language, you can use one of the built-in [language analyzers]({{site.url}}{{site.baseurl}}/query-dsl/analyzers/language-analyzers/) for your language of choice.
```

File: _search-plugins/index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+- Abstract OpenSearch queries into [search templates]({{site.url}}{{site.baseurl}}/search-plugins/search-template/).
+
+- Integrate machine learning (ML) language models into your search workloads with [neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/).
+
+- [Compare search results]({{site.url}}{{site.baseurl}}/search-plugins/search-relevance/) to tune search relevance.
+
+- Use a dataset that is fixed in time to paginate results with [Point in Time]({{site.url}}{{site.baseurl}}/search-plugins/point-in-time/).
+
+- [Paginate]({{site.url}}{{site.baseurl}}/opensearch/search/paginate) and [sort]({{site.url}}{{site.baseurl}}/opensearch/search/sort/) search results, [highlight] search terms, and use the [autocomplete]({{site.url}}{{site.baseurl}}/opensearch/search/autocomplete/) and [did-you-mean]({{site.url}}{{site.baseurl}}/opensearch/search/did-you-mean/) functionality.
```
Comment (vagimeli) [inline _search-plugins/index.md:27]: Should [highlight] include a hyperlink?

**Review by vagimeli — APPROVED** [review]

File: _dashboards/index.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+layout: default
+title: OpenSearch Dashboards
+nav_order: 1
+has_children: false
+---
+
+# OpenSearch Dashboards
+
+OpenSearch Dashboards is the user interface that lets you visualize your OpenSearch data and run and scale your OpenSearch clusters. Get started with OpenSearch Dashboards today!
```
Comment (hdhalter) [inline _dashboards/index.md:10]:
```suggestion
OpenSearch Dashboards is the user interface that lets you visualize your OpenSearch data, and run and scale your OpenSearch clusters.
```

File: _dashboards/index.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Getting started
+
+| Concept | Description | 
+|---------|-------------|
+| [OpenSearch quickstart]({{site.url}}{{site.baseurl}}/dashboards/quickstart-dashboards/) | Learn about the basic concepts and features in OpenSearch Dashboards. |
```
Comment (hdhalter) [inline _dashboards/index.md:16]: OpenSearch Dashboards Quickstart

File: _dashboards/index.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+| [Install and configure OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/index/) | Get started with OpenSearch Dashboards. | 
+| [Upgrade OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/upgrade-opensearch/index/) | Get the latest version of OpenSearch and OpenSearch Dashboards. |
```
Comment (hdhalter) [inline _dashboards/index.md:18]: I would remove this.

File: _dashboards/index.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+| Concept | Description | 
+|---------|-------------|
+| [Observability in OpenSearch Dashboards]({{site.url}}{{site.baseurl}}//observing-your-data/index/) | Observe, monitor, and secure data and improve performance across tools and workflows. |
+
+## DevTools and APIs
+
+| Concept | Description |
+|---------|-------------|
+| [Dev Tools]({{site.url}}{{site.baseurl}}/dashboards/dev-tools/index-dev/) | Learn about the main uses of the Dev Tools Console, running queries and API requests, and using integrated tools to inspect data. |
```
Comment (hdhalter) [inline _dashboards/index.md:32]: the link isn't working for me.

File: _dashboards/index.md (hunk 3 above)
Comment (hdhalter) [reply _dashboards/index.md:10]: Should we point them to the playground?

**Review by hdhalter — CHANGES_REQUESTED** [review]

File: _dashboards/index.md (hunk 6 above)
Comment (kolchfa-aws) [reply _dashboards/index.md:32]: I would remove "using integrated tools to inspect data":
```suggestion
| [Dev Tools]({{site.url}}{{site.baseurl}}/dashboards/dev-tools/index-dev/) | Learn how to run OpenSearch queries in an integrated console. |
```

**Review by hdhalter — APPROVED** [review]
Looks, good!

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Review complete for only the files listed in the Asana request. Thanks!

**Pull request merged**